### DISCUSSION TITLE
segmentation fault using jsdom

### DISCUSSION THREAD

**Discussion by bbraun92** [vitest-dev/vitest#6760] [Q&A] [answered]
Hey,

I'm running tests using a single-threaded setup and I use jsdom. There's some tests relying on canvas that cause segmentation fault issues running in that mode, most of the tests work fine.
Is there any easy way to configure vitest so that the tests not causing issues are run in a single thread, and the ones causing trouble run in a pool of forks? Does that even make sense?

Thank you

**Comment by AriPerkkio** [comment, marked as answer]
Yep, `canvas` doesn't work in `node:worker_threads` and requires `pool: 'forks'` in Vitest: https://github.com/vitest-dev/vitest/issues/2261#issuecomment-1893110542

You can use [`poolMatchGlobs`](https://vitest.dev/config/#poolmatchglobs) to run specific tests in `forks` pool:

```ts
import { defineConfig } from 'vitest/config'

export default defineConfig({
  test: {
    poolMatchGlobs: [
      ['some/path/to/testfile.test.ts', 'forks'],
    ],
  },
})
```

Or you could just run all tests in single fork instead of single thread, if that's what you want:

```ts
import { defineConfig } from 'vitest/config'

export default defineConfig({
  test: {
    poolOptions: { forks: { singleFork: true } },
  },
})
```

**Reply by bbraun92** [reply]
Thank you so much!

Right now my setup looks like

```
export default defineConfig({
  test: {
    poolOptions: {
      threads: {
        singleThread: true,
      },
    },
    poolMatchGlobs: [
        '**/test/unit/utils/my-test.js',
        'forks',
      ],
  }
})
```

`my-test.js` is the only test using `canvas` and if I remove it entirely my pipeline runs just fine. For some reason my pipeline still fails reporting `segmentation fault` with the updated setup. Can you mix `poolOptions` and `poolMatchGlobs` that way?

If I use

```
export default defineConfig({
  test: {
    pool: 'forks',
  }
})
```

The issue disappears. But I'd really like to only isolate the faulty test using forks as I understand threads should be more efficient usually.

**Reply by AriPerkkio** [reply]
What Vitest version are you using?

Yes, you can mix `poolOptions` and `poolMatchGlobs` like this. You can verify that the file is run correctly by adding `console.log('Is forks pool:', process.send != undefined);` in `my-test.js`.

**Reply by bbraun92** [reply]
I'm on 1.6.0, I'll see if I can verify it thank you!

**Reply by bbraun92** [reply]
`console.log('Is forks pool:', process.send != undefined);` logs Is forks pool: false 🤔

**Reply by AriPerkkio** [reply]
This should help:

```diff
    poolMatchGlobs: [
+     [
        '**/test/unit/utils/my-test.js',
        'forks',
+     ]
      ],
```

**Reply by bbraun92** [reply]
Oh man, that's a bit embarassing 🙈 
Yeah that fixed it, thank you very much!
